tutorials/deployment-with-docker.md

```diff
@@ -0,0 +1,119 @@
+
+# Production
+If you want to use NodeCG in production, it is recommended to build a own Docker Image to hold all your bundles and configuration.
+
+Example Dockerfile:
+```Dockerfile
+FROM nodecg/nodecg:v1.5.0
+
+RUN nodecg install [bundle-repo] && nodecg defaultconfig [bundle-name] && \
+    ... 
+    
+COPY --chown=nodecg:nodecg [nodecg-config-file] /opt/nodecg/cfg/nodecg.json
+COPY --chown=nodecg:nodecg [bundle-config-file] /opt/nodecg/cfg/[bundle].json
+```
+
+> Instead of the seperate `COPY` directives you could use only one:
+> 
+> ```Dockerfile
+> COPY --chown=nodecg:nodecg [config-dir]/* /opt/nodecg/cfg/
+> ```
+>
+> This will create only one layer. But requires you to organize all your configuration files in the same directory.
+> [More Information](https://docs.docker.com/engine/reference/builder/#copy)
+
+1. `FROM nodecg/nodecg:v1.5.0` Defines the parent image that should be used to build your Docker Image. We use `nodecg/nodecg` with the latest version tag `v1.5.0`. The parent image already include [NodeCG](https://nodecg.com) and [nodecg-cli](https://github.com/nodecg/nodecg-cli).
+
+2. `RUN nodecg install ... && nodecg defaultconfig ...` Installs your bundles and generates the default configs for them. (if you include the configuration later, you dont need to generate the default configuration)
+
+    > nodecg install (more information at [nodecg-cli](https://github.com/nodecg/nodecg-cli))
+    > 
+    > To install a bundle from Github, enter the owner and repository name:
+    >
+    > `nodecg install lange/lange-notify`
+    >
+    > ... to install a bundle from Bitbucket, enter the owner and repository name prefixed with bitbucket:
+    >
+    > `nodecg install bitbucket:username/repo-name`
+    >
+    > ... to install a bundle from any other git provider, enter the git URL:
+    >
+    > `nodecg install https://gitlab.com/username/repo-name.git`
+
+3. `COPY --chown=nodecg:nodecg [nodecg-config-file] /usr/src/app/cfg/nodecg.json` Copies the config file from your filesystem into the Docker Image and can be accessed from the container. (`--chown` is needed because the NodeCG process is run by the user `nodecg`)
+
+Now use Docker to build your image from the Dockerfile you just created:
+
+```sh
+docker build -t [your-image-name] .
+```
+
+And run it with:
+```sh
+docker run -p 9090:9090 [your-image-name]
+```
+
+Or create a docker-compose.yml like this:
+
+```yaml
+version: '3'
+services:
+  nodecg:
+    image: '[your-image-name]'
+    ports:
+      - '9090:9090'
+```
+
+Build a new docker image for your deployment has the advantage, that it is safer (because it is run as a non-root user), more consistent (because the container wont keep changes, except your explicitly tell it to).
+
+To upgrade NodeCG just build a new image with the new parent.
+Replace `FROM nodecg/nodecg:1.5.0` with `FROM nodecg/nodecg:[newer version]`
+
+## Caution: Public accessable NodeCG instances
```

Comment:
This is nothing to do with Docker.

Reply:
I added "Caution: Public accessable NodeCG instances" because in #422 there is a TLS/SSL-Termination Proxy

Reply:
Ohh I misunderstood, apologies.